velox/functions/sparksql/tests/SplitFunctionsTest.cpp

```diff
@@ -14,96 +14,350 @@
  * limitations under the License.
  */
 
-#include <optional>
+#include <gtest/gtest.h>
+#include "velox/common/base/tests/GTestUtils.h"
+#include "velox/expression/Expr.h"
+#include "velox/functions/Udf.h"
 #include "velox/functions/sparksql/tests/SparkFunctionBaseTest.h"
+#include "velox/parse/Expressions.h"
 
-namespace facebook::velox::functions::sparksql::test {
-
+using namespace facebook::velox;
+using namespace facebook::velox::exec;
+using namespace facebook::velox::functions::test;
 using namespace facebook::velox::test;
+
+namespace facebook::velox::functions::sparksql::test {
 namespace {
 
 class SplitTest : public SparkFunctionBaseTest {
  protected:
-  void testSplitCharacter(
-      const std::vector<std::optional<std::string>>& input,
-      std::optional<char> pattern,
-      const std::vector<std::optional<std::vector<std::string>>>& output);
+  /// Method runs the given split function, f.e. split(C0, C1), where C0 is the
+  /// input column and the C1 is delimiter column.
+  /// Encoding arguments control what kind of vectors we should create for the
+  /// function arguments.
+  /// limit should be set to the corresponding limit, if query contains limit
+  /// argument (C2).
+  VectorPtr run(
```

Comment:
Function `testEncodings` could be used to test different encodings.

Reply:
the `testEncondings` is not good enough for our tests since it only test with inputs all has one encoding , for example the inputs all are `flat`, or `dictionary` or `constant`. 

But our code path also need verify for each input has different encoding as `string` is `flat` encoding, others (delim, pattern) is `constant`  etc to verify two code path
https://github.com/gayangya/velox/blob/gayangya/split_refactor/velox/functions/sparksql/SplitFunctions.cpp#L52C1-L53C52 and https://github.com/gayangya/velox/blob/gayangya/split_refactor/velox/functions/sparksql/SplitFunctions.cpp#L52C1-L53C52

This part also follow similar way of how presto `splitTest` does - https://github.com/gayangya/velox/blob/gayangya/split_refactor/velox/functions/prestosql/tests/SplitTest.cpp#L42C1-L45C39

Reply:
@gaoyangxiaozhu Shall we extend the testEncodings API by adding a testMixed function which generates mixed encodings and verify the result? I'm glad to support that in a separate PR.